src/main/kotlin/script/birthday/BirthdayScript.kt

```diff
@@ -0,0 +1,173 @@
+                val result = getNextBirthdaysIn7DaysFromAllUsers()
+                val response = when {
+                    result.isNotEmpty() -> "Birthdays:\n" + result.joinToString(separator = "\n")
+                    else -> "No birthday in the next 7 days"
+                }
+                ctx.say(response)
+            }
+
+            "show all birthdays" -> {
+                ctx.say("Birthdays:\n" + getBirthdaysFromAllUsers().joinToString(separator = "\n"))
+            }
+
+            "notify users about birthdays" -> {
+                notifyUserForIncomingBirthdays(ctx)
+            }
+        }
+    }
+
+    private fun getBirthdaysFromAllUsers(): List<String> {
+        val users = getAllUser()
+
+        return users.map { user ->
+            val userName = user.id.usernameString
+            val daysUntilMsg = user.birthdate?.let { it1 -> calculateDaysUntilBirthday(currentDate.toLocalDate(), it1) }
+            daysUntilMsg?.let { message -> "$userName hat am ${getBirthdate(user.birthdate)} (in $message Tagen) Geburtstag" } ?: "Kein Geburtstag hinterlegt"
+        }
+    }
+
+    private fun getNextBirthdaysIn7DaysFromAllUsers(): List<String> {
+        val users = getAllUser()
+
+        return users.mapNotNull {
+            val userName = it.id.usernameString
+            val daysUntilMsg = when (val daysUntilBirthday = it.birthdate?.let { it1 -> calculateDaysUntilBirthday(currentDate.toLocalDate(), it1) }) {
+                7 -> "in einer Woche"
+                3 -> "in $daysUntilBirthday Tagen"
+                1 -> "morgen"
+                0 -> "heute"
+                else -> null
+            }
+            daysUntilMsg?.let { "$userName hat $daysUntilMsg Geburtstag" }
+        }
+    }
+
+    private fun List<SelectUsersForTeam>.getNextBirthdaysIn7Days(): List<String> {
+        return mapNotNull {
+            val userName = it.id.usernameString
+            val daysUntilMsg = when (val daysUntilBirthday = it.birthdate?.let { it1 -> calculateDaysUntilBirthday(currentDate.toLocalDate(), it1) }) {
+                7 -> "in einer Woche"
+                3 -> "in $daysUntilBirthday Tagen"
+                1 -> "morgen"
+                0 -> "heute"
+                else -> null
+            }
+            daysUntilMsg?.let { "$userName hat $daysUntilMsg Geburtstag" }
+        }
+    }
+
+    private fun notifyUserForIncomingBirthdays(ctx: EventContext) {
+        val users = getAllUser()
+        users.forEach { user ->
+            val nextBirthdaysIn7Days = user.getBirthdaysForUser()
+
+            val response = if (nextBirthdaysIn7Days.isNotEmpty()) {
+                buildString {
+                    appendLine("Geburtstage:")
+                    nextBirthdaysIn7Days.forEach { it ->
+                        appendLine(it)
+                    }
+                }
+            } else {
+                "Keine anstehenden Geburtstage"
+            }
+            ctx.client().chatPostMessage {
```

Comment:
So einen 1:1 Chat zu starten würde ich gerne allgemein in eine Helper-Methode auslagern 🙂